src/main/java/tasks/TaskList.java

```diff
@@ -0,0 +1,172 @@
+     * @return Delete task message
+     */
+    public String deleteTask(String taskNum){
+        int index = Integer.parseInt(taskNum) - 1;
+        Task task = this.remove(index);
+        return "Noted. I've removed this task:\n" + task.toString()
+                + "\nNow you have " + this.getSize() + " tasks in the list";
+    }
+
+    /**
+     * Returns list of tasks in String form
+     * @return List of tasks in String form
+     */
+    public String listTasks(){
+        String list = "Here are the tasks in your list:\n";
+        for (int i = 0; i < this.size(); i++){
+            list += (i + 1) + "." + this.get(i).toString() + "\n";
+        }
+        return list;
+    }
+
+    /**
+     * Returns add task message.
+     * Generates new Todo object and adds it to itself.
+     * @param desc Description of the task
+     * @return Add task message
+     */
+    public String addTask(String desc){
+        Task todo = new Todo(desc);
+        this.add(todo);
+        return addTaskMsg(todo);
+    }
+
+    /**
+     * Returns add task message.
+     * Generates new Deadline object and adds it to itself.
+     * @param desc Description of the task
+     * @return Add task message
+     */
+    public String addTask(String desc, String by){
+        Task deadline = new Deadline(desc, by);
+        this.add(deadline);
+        return addTaskMsg(deadline);
+    }
+    /**
+     * Returns add task message.
+     * Generates new Event object and adds it to itself.
+     * @param desc Description of the task
+     * @return Add task message
+     */
+    public String addTask(String desc, String from, String to){
+        Task event = new Event(desc, from, to);
+        this.add(event);
+        return addTaskMsg(event);
+    }
+
+    /**
+     * Returns a message about the newly added task
+     * @param task Task object to be added
+     * @return String message about task
+     */
+    private String addTaskMsg(Task task){
+        int totalTasks = this.getSize();
+        return "Got it, I have added this task:\n" + task.toString() + "\n" +
+        "Now you have " + totalTasks + " tasks in the list.";
+    }
+
+    /**
+     * Returns mark task message.
+     * Marks targeted task as completed.
+     * @param index Task index
+     * @return Mark task message
+     */
+    public String markTask(int index){
+        Task task = this.get(index);
+        task.updateState();
+        return "Nice I have marked this task as done\n" + task.toString();
+    }
+    /**
+     * Returns unmark task message.
+     * Marks targeted task as not complete.
+     * @param index Task index
+     * @return Unmark task message
+     */
+    public String unmarkTask(int index){
+        Task task = this.get(index);
+        task.updateState();
+        return "Ok I have marked as not done\n" + task.toString();
+    }
+
+    /**
+     * Returns size of this object
+     * @return size of this object
+     */
+    public int getSize(){
```

Comment:
I am not really sure what this method is used for? Could probably just use `this.size()` directly.